crates/doc/src/builder.rs

```diff
@@ -99,20 +110,39 @@ impl DocBuilder {
             return Ok(())
         }
 
-        let documents = sources
+        let library_sources = self.libraries.iter().flat_map(source_files_iter).collect::<Vec<_>>();
+
+        let combined_sources = sources
+            .iter()
+            .map(|path| (path, false))
+            .chain(library_sources.iter().map(|path| (path, true)))
+            .collect::<Vec<_>>();
+
+        let documents = combined_sources
             .par_iter()
             .enumerate()
-            .map(|(i, path)| {
+            .map(|(i, (path, from_library))| {
+                let path = *path;
+                let from_library = *from_library;
+
                 // Read and parse source file
                 let source = fs::read_to_string(path)?;
-                let (mut source_unit, comments) =
-                    solang_parser::parse(&source, i).map_err(|diags| {
-                        eyre::eyre!(
+                let parser_result = solang_parser::parse(&source, i);
+
+                if parser_result.is_err() {
```

Comment:
I'd like this as a match statement insted that assigns let (mut source_unit, comments) =

so we can get rid of the unwraps